.github/workflows/release-plugin.yml

```diff
@@ -1,33 +1,233 @@
+          fetch-depth: 0
+
+      - name: Validate if the branch is built by checking if the vendor/autoload.php file exists
+        run: |
+          if [[ ! -f vendor/autoload.php ]]; then
+            echo "The branch '${{ env.SOURCE_REF }}' is not a built branch."
+            exit 1
+          fi
+
+      - name: Checks if the version matches the version in wp-parsely.php and package.json
+        if: ${{ env.DRY_RUN == false }}
+        run: |
+          PHP_VERSION=$(grep -E "^ \* Version:" wp-parsely.php | awk '{print $3}')
+          JSON_VERSION=$(jq -r '.version' package.json)
+          if [[ "${{ env.VERSION }}" != "${PHP_VERSION}" ]]; then
+            echo "Version '${{ env.VERSION }}' does not match the version in wp-parsely.php"
+            echo "Did you mean '${PHP_VERSION}'?"
+            exit 1
+          fi
+          if [[ "${{ env.VERSION }}" != "${JSON_VERSION}" ]]; then
+              echo "Version '${{ env.VERSION }}' does not match the version in package.json."
+              echo "Did you mean '${JSON_VERSION}'?"
+              exit 1
+          fi
+
+      - name: Checks if the version is in the CHANGELOG.md file
+        run: |
+          if ! grep -q "## \[${{ env.VERSION }}\]" CHANGELOG.md; then
+            echo "Version '${{ env.VERSION }}' is not in the CHANGELOG.md file."
+            exit 1
+          fi
+
+      - name: Checks if the version was already released
+        if: ${{ env.DRY_RUN == false }}
+        run: |
+          if git tag --list | grep -q "${{ env.VERSION }}"; then
+            echo "Version '${{ env.VERSION }}' has already been released."
+            exit 1
+          fi
+
+
+  integration_tests:
+    name: Integration Tests
+    needs: validate_version
+    if: ${{ github.event.inputs.skip_tests == 'false' }}
+    uses: ./.github/workflows/integration-tests.yml
+
+  e2e_tests:
+    name: End-to-end Tests
+    needs: validate_version
+    if: ${{ github.event.inputs.skip_tests == 'false' }}
+    uses: ./.github/workflows/e2e-tests.yml
+
+  tag_and_release:
+    name: Tag and Release
+    needs: [ validate_version, integration_tests, e2e_tests ]
+    if: ${{ !failure() && !cancelled() }}
+    runs-on: ubuntu-latest
+    outputs:
+      tag_name: ${{ steps.tag.outputs.tag_name }}
+    steps:
+      - name: Configure Git
+        run: |
+          git config --global user.name 'github-actions[bot]'
+          git config --global user.email 'github-actions[bot]@users.noreply.github.com'
+
+      - name: Checkout the specific branch/ref
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ env.SOURCE_REF }}
+          fetch-depth: 0
+
+      - name: Tag the release
+        id: tag
+        run: |
+          TAG_NAME="${{ env.VERSION }}"
+          if ${{ env.DRY_RUN }}; then
+            TAG_NAME="dry-run-${{ env.VERSION }}"
+            # If the tag already exists, delete the tag
+            if git tag --list | grep -q "${TAG_NAME}"; then
```

Comment:
Is there any possibility here that we could delete a tag that we don't want to delete?

Reply:
Very hard, since it will only remove a tag prefixed with `dry-run-` (`dry-run-3.17.0` for example).